cloud/docker_client.go

```diff
@@ -396,8 +402,14 @@ func (c *dockerClientImpl) GetDockerLogs(ctx context.Context, h *host.Host, cont
 		return nil, errors.Wrapf(err, "Error copying stream for container %s", containerID)
 	}
 
-	res.OutReader = tempout
-	res.ErrReader = temperr
+	res.OutStr = tempout.String()
+	res.ErrStr = temperr.String()
+
+	container, err := c.GetContainer(ctx, parent, containerID)
+	if err != nil || container == nil {
+		return nil, errors.Wrapf(err, "Error getting container %s", containerID)
```

Comment:
this will confusingly return nil, nil, if err == nil and container != nil

Reply:
I assume you mean that if err == nil and container == nil then errors.Wrapf will return nil, which is a good catch. But if err == nil and container != nil we'll skip past this if statement, set res.IsRunning to the container state, and return res, nil.